runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala

```diff
@@ -197,43 +221,47 @@ object AkkaHttpClientUtils {
                             trailerPromise.trySuccess(immutable.Seq.empty)
                           }))
-                        // This never adds any data to the stream, but makes sure it fails with the correct error code if applicable
-                        .concat(Source
-                          .maybe[ByteString]
-                          .mapMaterializedValue(promise => promise.completeWith(completionFuture.map(_ => None))))
-                        // Make sure we continue reading to get the trailing header even if we're no longer interested in the rest of the body
-                        .via(new CancellationBarrierGraphStage)
-                        .via(reader.dataFrameDecoder)
-                        .map(data => deserializer.deserialize(data))
-                        .mapMaterializedValue(_ =>
-                          Future.successful(new GrpcResponseMetadata() {
-                            override def headers: akka.grpc.scaladsl.Metadata =
-                              new HeaderMetadataImpl(response.headers)
-
-                            override def getHeaders(): akka.grpc.javadsl.Metadata =
-                              new JavaMetadataImpl(new HeaderMetadataImpl(response.headers))
-
-                            override def trailers: Future[akka.grpc.scaladsl.Metadata] =
-                              trailerPromise.future.map(new HeaderMetadataImpl(_))
-
-                            override def getTrailers(): CompletionStage[akka.grpc.javadsl.Metadata] =
-                              trailerPromise.future
-                                .map[akka.grpc.javadsl.Metadata](h => new JavaMetadataImpl(new HeaderMetadataImpl(h)))
-                                .toJava
-                          }))
+                    case Strict(_, data) =>
+                      trailerPromise.success(immutable.Seq.empty)
+                      Source.single[ByteString](data)
                     case _ =>
                       response.entity.discardBytes()
                       throw mapToStatusException(response, Seq.empty)
                   }
-                case Failure(e) =>
-                  Source.failed[O](e).mapMaterializedValue(_ => Future.failed(e))
-              }
+                responseData
+                  // This never adds any data to the stream, but makes sure it fails with the correct error code if applicable
+                  .concat(
+                    Source
+                      .maybe[ByteString]
+                      .mapMaterializedValue(promise => promise.completeWith(completionFuture.map(_ => None))))
+                  // Make sure we continue reading to get the trailing header even if we're no longer interested in the rest of the body
+                  .via(new CancellationBarrierGraphStage)
+                  .via(reader.dataFrameDecoder)
+                  .map(data => deserializer.deserialize(data))
```

Comment:
To go with the same style as the line above?

```suggestion
                  .map(deserializer.deserialize)
```